src/main/java/com/synopsys/integration/alert/channel/azure/boards/web/AzureBoardsCustomEndpoint.java

```
@@ -22,37 +22,118 @@
+            Optional<String> alertServerUrl = alertProperties.getServerUrl();
+
+            if (!alertServerUrl.isPresent()) {
+                return new OAuthEndpointResponse(HttpStatus.BAD_REQUEST.value(), false, "", "Could not determine the alert server url for the callback.");
+            }
+            String authUrl = createAuthURL(clientId.get(), alertServerUrl.get());
+            logger.info("Authenticating Azure OAuth URL: " + authUrl);
+            //TODO add code to check if Alert has already been authorized to set the authenticated flag.
+
+            return new OAuthEndpointResponse(HttpStatus.OK.value(), false, authUrl, "");
+
+        } catch (Exception ex) {
+            logger.error("Error activating Azure Boards", ex);
+            return new OAuthEndpointResponse(HttpStatus.INTERNAL_SERVER_ERROR.value(), false, "", "Error activating azure oauth.");
+        }
+    }
+
+    private FieldAccessor createFieldAccessor(FieldModel fieldModel) {
+        Map<String, ConfigurationFieldModel> fields = new HashMap<>();
+        try {
+            fields.putAll(modelConverter.convertToConfigurationFieldModelMap(fieldModel));
+            // check if a configuration exists because the client id is a sensitive field and won't have a value in the field model if updating.
+            if (StringUtils.isNotBlank(fieldModel.getId())) {
+                Optional<ConfigurationModel> configurationFieldModel = configurationAccessor.getConfigurationById(Long.valueOf(fieldModel.getId()));
+                configurationFieldModel.ifPresent(model -> fields.putAll(model.getCopyOfKeyToFieldMap()));
+            }
+        } catch (AlertDatabaseConstraintException ex) {
+            logger.error("Error creating field acessor for Azure authentication", ex);
+        }
+        return new FieldAccessor(fields);
     }
 
+    private String createAuthURL(String clientId, String alertServerUrl) {
+        StringBuilder authUrlBuilder = new StringBuilder(300);
```

Is the 300 here the max length? We should probably throw an exception if we violate that constraint if it's required by Microsoft.

---

The 300 isn't a max length.  It's a best guess for initial capacity.  My test URL was 237 characters long. This is so the StringBuilder allocates a string with enough capacity initially instead of resizing more than once with the append calls.  it will resize dynamically like an ArrayList.  So this just makes it perform a little better. The string underneath is backed by a character array. So the character array will be created with a good enough size to avoid resizing during the append calls.